File: playground/sum_inv_n_square.R

print("Series 1/n^2")

my_inv_square <- function(x){
  tmp <- 1./as.double(x*x)
  res <- if(is.na(tmp)) 0  else tmp 
  return(res)
}


my_sum_inv_square <- function(N){
  res <- Reduce(function (x, y) x +  my_inv_square(y), x = 1:N, accumulate = FALSE)
  return(res)
}

warnings()  

print("theo")
theo <- pi*pi/6.
theo

my_res <- my_sum_inv_square(200000) 
print("my res")
my_res
print(paste0("Delta R =", my_res - theo))

File: playground/sum_inv_n_square.jl

println("Series 1/n^2")

# Print the multiplication talbe of 9
#list = 1:10
#for i in list
#    println("9 * $i = ",9*i)
#end

my_sum_inv_square(N) = sum(1. / (i*i) for i in 1:N)

theo = Float64(pi*pi / 6)
res = my_sum_inv_square(200000) # my_sum(1000000000)

println("res  = ", res)
println("pi*pi/6  = ", theo)
println("Delta  = ", res-theo)


